File: src/main/typescript/types/Organization.ts

export interface Address {
    street: string
    number: string
    zip: string
    place: string
    country: string
}

export interface OrganizationDescriptor {
    id: number
    name: string
    address: Address
}

export interface Organization extends OrganizationDescriptor {
    logo: string
    phone: string
    email: string
    fax: string
    website: string
}

export const defaultOrganization: Organization = {
    id: 0,
    name: undefined,
    logo: undefined,
    phone: undefined,
    email: undefined,
    address: {
        street: undefined,
        number: undefined,
        zip: undefined,
        place: undefined,
        country: undefined
    },
    fax: undefined,
    website: undefined
}

export const toAddressString = (address: Address) => {
    return `${address.street} ${address.number}, ${address.zip} ${address.place}, ${address.country}`
}

File: src/main/java/orakuma/demo/model/Organization.java
package orakuma.demo.model;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Lob;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "organizations")
public class Organization implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = -1128572054634030032L;

	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "org_gen")
	@SequenceGenerator(name = "org_gen", sequenceName = "org_seq")
	private long id;

	@Column(name = "name")
	private String name;

	@Lob
	@Column(name = "logo")
	private byte[] logo;

	@Column(name = "address")
	@Embedded
	private Address address;

	@Column(name = "email")
	private String email;

	@Column(name = "phone")
	private String phone;

	@Column(name = "fax")
	private String fax;

	@Column(name = "website")
	private String website;
	
	@Column(name = "cellphone")
	private String cellphone;
}